Client/Frontend/TabBar/SearchViewController.swift

```diff
@@ -13,10 +14,36 @@ protocol SearchViewControllerDelegate: class {
 
 class SearchViewController: UIViewController {
     weak var delegate: SearchViewControllerDelegate?
-    var searchEngines: [OpenSearchEngine]?
-
-    private var searchText = ""
     private var tableView = UITableView()
+    private var sortedEngines = [OpenSearchEngine]()
+    private var suggestClient: SearchSuggestClient?
+    private var searchSuggestions = [String]()
+
+    var searchEngines: SearchEngines? {
+        didSet {
+            if let searchEngines = searchEngines {
+                // Show the default search engine first.
+                sortedEngines = searchEngines.list.sorted { engine, _ in engine === searchEngines.defaultEngine }
+                suggestClient = SearchSuggestClient(searchEngine: searchEngines.defaultEngine)
+            } else {
+                sortedEngines = []
+                suggestClient = nil
+            }
+            requerySuggestClient()
+
+            // Reload the tableView to show the new list of search engines.
+            tableView.reloadData()
+        }
+    }
+
+    var searchQuery: String = "" {
+        didSet {
```

Comment:
This is nice. I had previously used a setter/getter in combination with a shadowed 'real' variable. Much easier with `didSet`.